modules/dnn/test/test_encoding_bpe.cpp

```diff
@@ -0,0 +1,305 @@
+#include "test_precomp.hpp"
+#include "../src/tokenizer/core_bpe.hpp"
+#include "../src/tokenizer/encoding.hpp"
+#include "../src/tokenizer/utils.hpp"
+#include "../include/opencv2/dnn/tokenizer.hpp"
+#include "../src/tokenizer/gpt2_tokenizer_fast.hpp"
```

Comment:
Avoid including headers files here.

The definition of newly added methods or classes should be put in [dnn.hpp](https://github.com/opencv/opencv/blob/4.x/modules/dnn/include/opencv2/dnn/dnn.hpp).